controller/src/controller/subsystems/instrument_comm.py

```diff
@@ -584,6 +652,17 @@ async def _process_command_response(self, packet_type: int, response_data: bytes
                 if self._firmware_update_manager is None:
                     raise NotImplementedError("_firmware_update_manager should never be None here")
                 await self._firmware_update_manager.update(command, response_data)
+            case "check_connection_status":
+                prev_command_info["status"] = response_data[0]
+                self._system_in_offline_mode = response_data[0] == ConnectionStatuses.HEADLESS.value
+                self._offline_state_change.set()
+                if self._system_in_offline_mode:
+                    logger.info("Starting up in offline mode")
+            case "end_offline_mode":
+                parsed_stim_dict = parse_end_offline_mode_bytes(response_data)
+                prev_command_info |= {"stim_state": parsed_stim_dict}
```

Comment:
I think we could just put all the keys in the top level of the dict
`prev_command_info |= parse_end_offline_mode_bytes(response_data)`

Reply:
yeah honestly the response I didn't was to avoid having the add all the keys in the match statement in system_monitor

Reply:
You don't need to add all the keys, it will only match on the keys in the statement

Reply:
It returned this error:
```
Traceback (most recent call last):
  File "/Users/lucipak/Documents/CuriBio/stingray-controller/controller/src/controller/main_systems/system_monitor.py", line 55, in run
    await wait_tasks_clean(tasks, error_msg=ERROR_MSG)
  File "/Users/lucipak/Documents/CuriBio/stingray-controller/controller/src/controller/utils/aio.py", line 26, in wait_tasks_clean
    raise exc
  File "/Users/lucipak/Documents/CuriBio/stingray-controller/controller/src/controller/utils/aio.py", line 39, in clean_up_tasks
    await task
  File "/Users/lucipak/Documents/CuriBio/stingray-controller/controller/src/controller/main_systems/system_monitor.py", line 353, in _handle_comm_from_instrument_comm
    raise NotImplementedError(f"Invalid communication from InstrumentComm: {invalid_comm}")
NotImplementedError: Invalid communication from InstrumentComm: {'command': 'end_offline_mode', 'system_dormant_timestamp': 16203375319481057280, 'stim_active': True, 'last_stim_scheduled_start_timestamp': 16203375319481057280, 'stimulator_statuses': bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'), 'stim_info': {'protocols': [{'stimulation_type': 'C', 'run_until_stopped': True, 'subprotocols': [{'type': 'loop', 'num_iterations': 1, 'subprotocols': [{'type': 'monophasic', 'phase_one_duration': 10000, 'phase_one_charge': 100000, 'postphase_interval': 90000, 'num_cycles': 10}]}], 'protocol_id': 'A'}], 'protocol_assignments': {'A1': 'A', 'B1': 'A', 'C1': 'A', 'D1': 'A', 'A2': None, 'B2': None, 'C2': None, 'D2': None, 'A3': None, 'B3': None, 'C3': None, 'D3': None, 'A4': None, 'B4': None, 'C4': None, 'D4': None, 'A5': None, 'B5': None, 'C5': None, 'D5': None, 'A6': None, 'B6': None, 'C6': None, 'D6': None}}}
```